src/utils/data-stream.ts

```diff
@@ -0,0 +1,48 @@
+import { Encoder } from './encoder.js';
+import { Readable } from 'readable-stream';
```

Comment:
so on slack I thought we were using `ReadableStream` the global. however this seems to be a package, presumably it's a polyfill for/from node envs? i'm not sure how this is going to work on alternative runtimes. but presumably since it's here, it works with web? i think I need some more context to understand

Reply:
You are right, it is a package: https://www.npmjs.com/package/readable-stream

I did not dig in too much on how they work their magic, I only know that:
1. It's meant to work in browsers by design
1. DWN tests (now run on chrome, firefox, and safari) confirms above
1. polyfill is not required from v4+
1. got 100M+ downloads a week

The above were good enough of an indicator for me to depend on it in this PR. But not sure about any other gotchas specific to RN etc.